app/controllers/api/goals_controller.rb

```diff
@@ -12,6 +12,26 @@ def create
     end
   end
 
+  def update
+    goal = Goal.find(params[:id])
```

Comment:
Consider limiting this query by using a current_user-scoped query to prevent users updating each other's goals:

```diff
- goal = Goal.find(params[:id])
+ goal = current_user.goals.find(params[:id])
```